packages/SwingSet/src/kernel/kernel.js

```diff
@@ -403,6 +405,11 @@ export default function buildKernel(
     const kd = harden(['message', target, msg]);
     // eslint-disable-next-line no-use-before-define
     const vd = vatWarehouse.kernelDeliveryToVatDelivery(vatID, kd);
+
+    if (vatInfo.enablePipelining && msg.result) {
```

Comment:
This is probably the only good place to perform the requeuing, but it'd be nice if there were a way to do it in the translator, at the same time we do the `setDecider`. But `vatInfo.enablePipelining` isn't very available there, which suggests we have a funny split between where we keep track of per-vat runtime-ish things like `enablePipelining`, and per-message / per-clist-entry state. It would certainly be wrong to make `kernelKeeper.setDecider()` do the re-queueing, that's too much for that method, but maybe we should plan a future refactoring that creates a "changePromiseDeciderAndRequeue" method somewhere.

No changes recommended now.